consensus/src/round_manager.rs

```diff
@@ -227,16 +227,16 @@ impl RoundManager {
             .proposer_election
             .is_valid_proposer(self.proposal_generator.author(), new_round_event.round)
         {
-            let proposal_msg = Box::new(self.generate_proposal(new_round_event).await?);
+            let proposal_msg = self.generate_proposal(new_round_event).await?;
             let mut network = self.network.clone();
             #[cfg(feature = "failpoints")]
             {
-                self.attempt_to_inject_reconfiguration_error(&proposal_msg)
```

Comment:
this was unconditionally enabled, now is behind "consensus::inject_reconfiguration_error"

I searched for other occurrences of #[cfg(feature = "failpoints")], and this seems to be the only broken one.

Reply:
oh i remember this one, I was lazy and intentionally enabled it for smoke tests (implicitly) lol, good catch